cookbook/core/control_flow/gate_nodes.py

```diff
@@ -0,0 +1,241 @@
+#   attaches the custom title to the report.
+#
+# As mentioned in the beginning of this page, this construct can be used for
+# selecting the best-performing model in cases where there isn't a clear single
+# metric to determine the best model, or if you're doing data labeling using
+# a Flyte workflow.
+#
+# Continue executions with ``approve``
+# ====================================
+#
+# Finally, the :py:func:`~flytekit.approve` gate node allows you to wait on
+# an explicit approval signal before continuing execution. Going back to our
+# report-publishing use case, suppose that we want to block the publishing of
+# a report for some reason (e.g. they don't appear to be valid):
+
+# %%
+# approve based on a task output
+from flytekit import approve
+
+
+@workflow
+def reporting_with_approval_wf(data: typing.List[float]) -> dict:
+    report = create_report(data=data)
+    title_input = wait_for_input("title", timeout=timedelta(hours=1), expected_type=str)
+    final_report = finalize_report(report=report, title=title_input)
+
+    # approve the final report, where the output of approve is the final_report
+    # dictionary.
+    return approve(final_report, "approve-final-report", timeout=timedelta(hours=2))
+
+
+# %%
+# You can also use the output of the ``approve`` function as a promise, feeding
+# it to a subsequent task. Let's create a version of our report-publishing
+# workflow where the approval happens after ``create_report``:
+
+@workflow
+def approval_as_promise_wf(data: typing.List[float]) -> dict:
+    report = create_report(data=data)
+    title_input = wait_for_input("title", timeout=timedelta(hours=1), expected_type=str)
+
+    # wait for report to 
+    report >> title_input
+
+    final_report = finalize_report(
+        report=approve(report, "raw-report-approval", timeout=timedelta(hours=2)),
+        title=title_input
+    )
+    return final_report
+
+
+# %%
+# Gate nodes with conditionals
+# ============================
+#
+# The gate node constructs by themselves are useful, but they become even more
+# useful when we combine them with other Flyte constructs, like :ref:`conditionals <conditional>`.
+#
+# To illustrate this, let's extend the report-publishing use case so that we
+# produce and "invalid report" output in case we don't approve the final report:
+
+from flytekit import conditional
+
+@task
+def invalid_report() -> dict:
+    return {"invalid_report": True}
+
+@workflow
+def gate_node_with_conditional_wf(data: typing.List[float]) -> dict:
+    report = create_report(data=data)
+    title_input = wait_for_input("title", timeout=timedelta(hours=1), expected_type=str)
+    final_report = finalize_report(report=report, title=title_input)
+
+    # use wait_for_input instead of approve here such that the type of the
+    # approved promise is a boolean, not a dictionary
+    review_passed = wait_for_input("review-passes", timeout=timedelta(hours=2), expected_type=bool)
+    final_report >> review_passed
```

Comment:
I don't think this line is required here as `conditional` runs the `final_report` task if `review_passed` is true.